packages/explat-client/src/test/create-explat-client.ts

```diff
@@ -0,0 +1,697 @@
+		spiedMonotonicNow.mockImplementationOnce(
+			() => validExperimentAssignment.retrievedTimestamp + 1000
+		);
+		await expect(
+			client.loadExperimentAssignment( validExperimentAssignment.experimentName )
+		).resolves.toEqual( validExperimentAssignment );
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `Array []` );
+	} );
+	it( `Invalid experimentName: should return fallback and log`, async () => {
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		const client = createExPlatClient( createMockedConfig() );
+		await expect(
+			client.loadExperimentAssignment( 'the-invalid-experiment-name' )
+		).resolves.toEqual( {
+			experimentName: 'fallback_experiment_assignment',
+			retrievedTimestamp: validExperimentAssignment.retrievedTimestamp,
+			ttl: 60,
+			variationName: null,
+			isFallbackExperimentAssignment: true,
+		} );
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `
+		Array [
+		  Array [
+		    Object {
+		      "experimentName": "the-invalid-experiment-name",
+		      "message": "Invalid experimentName: the-invalid-experiment-name",
+		    },
+		  ],
+		]
+	` );
+	} );
+	it( `[developmentMode] Invalid experimentName: should throw`, async () => {
+		mockFetchExperimentAssignmentToMatchExperimentAssignment( validExperimentAssignment );
+		const client = createExPlatClient( createMockedConfig( DEVELOPMENT_MODE ) );
+		await expect(
+			client.loadExperimentAssignment( 'the-invalid-experiment-name' )
+		).rejects.toThrow( 'Invalid experimentName: the-invalid-experiment-name' );
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `
+		Array [
+		  Array [
+		    Object {
+		      "experimentName": "the-invalid-experiment-name",
+		      "message": "Invalid experimentName: the-invalid-experiment-name",
+		    },
+		  ],
+		]
+	` );
+	} );
+	it( `Could not fetch ExperimentAssignment: should store and return fallback, and log`, async () => {
+		mockedFetchExperimentAssignment.mockImplementationOnce(
+			() => new Promise( ( _res, rej ) => rej( new Error( 'some-error-123' ) ) )
+		);
+		const client = createExPlatClient( createMockedConfig() );
+		const firstNow = Date.now();
+		spiedMonotonicNow.mockImplementationOnce( () => firstNow );
+		const secondNow = firstNow + 1;
+		spiedMonotonicNow.mockImplementationOnce( () => secondNow );
+		await expect(
+			client.loadExperimentAssignment( validExperimentAssignment.experimentName )
+		).resolves.toEqual( {
+			experimentName: validExperimentAssignment.experimentName,
+			retrievedTimestamp: secondNow,
+			ttl: 60,
+			variationName: null,
+			isFallbackExperimentAssignment: true,
+		} );
+		expect( mockedLogError.mock.calls ).toMatchInlineSnapshot( `
+		Array [
+		  Array [
+		    Object {
+		      "experimentName": "experiment_name_a",
+		      "message": "some-error-123",
+		    },
+		  ],
+		]
+	` );
+	} );
+
+	it( `[developmentMode] Could not fetch ExperimentAssignment: should throw`, async () => {
```

Comment:
Not critical, but this file may be easier to follow if we split the development mode tests to nested `describe`s. That is:

```typescript
describe( 'ExPlatClient.loadExperimentAssignment single-use', () => {
	describe( 'in production mode, () => {
		// ... production mode tests ...
	} );

	describe( 'in development mode, () => {
		// ... development mode tests ...
	} );
} );
```

Reply:
I was thinking about this but I like that the development and production tests are next to each other - it is easier to compare behaviour, makes refactoring a bit more obvious, and makes an omissions of a corresponding developer test clear.